core/src/test/java/org/elasticsearch/rest/DeprecationHttpIT.java

```diff
@@ -0,0 +1,200 @@
+import static org.elasticsearch.rest.plugins.TestDeprecationHeaderRestAction.TEST_NOT_DEPRECATED_SETTING;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.hasStatus;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.hasSize;
+
+/**
+ * Tests {@code DeprecationLogger} uses the {@code ThreadContext} to add response headers.
+ */
+public class DeprecationHttpIT extends ESIntegTestCase {
+
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal) {
+        return Settings.builder()
+                .put(super.nodeSettings(nodeOrdinal))
+                .put("force.http.enabled", true)
+                // change values of deprecated settings so that accessing them is logged
+                .put(TEST_DEPRECATED_SETTING_TRUE1.getKey(), ! TEST_DEPRECATED_SETTING_TRUE1.getDefault(Settings.EMPTY))
+                .put(TEST_DEPRECATED_SETTING_TRUE2.getKey(), ! TEST_DEPRECATED_SETTING_TRUE2.getDefault(Settings.EMPTY))
+                // non-deprecated setting to ensure not everything is logged
+                .put(TEST_NOT_DEPRECATED_SETTING.getKey(), ! TEST_NOT_DEPRECATED_SETTING.getDefault(Settings.EMPTY))
+                .build();
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return pluginList(TestDeprecationPlugin.class);
+    }
+
+    /**
+     * Attempts to do a scatter/gather request that expects unique responses per sub-request.
+     */
+    @AwaitsFix(bugUrl = "")
+    public void testUniqueDeprecationResponsesMergedTogether() throws IOException {
+        final String[] indices = new String[randomIntBetween(2, 5)];
+
+        // add at least one document for each index
+        for (int i = 0; i < indices.length; ++i) {
+            indices[i] = "test" + i;
+
+            // create indices with a single shard to reduce noise; the query only deprecates uniquely by index anyway
+            assertTrue(prepareCreate(indices[i]).setSettings(Settings.builder().put("number_of_shards", 1)).get().isAcknowledged());
+
+            int randomDocCount = randomIntBetween(1, 2);
+
+            for (int j = 0; j < randomDocCount; ++j) {
+                index(indices[i], "type", Integer.toString(j), "{\"field\":" + j + "}");
+            }
+        }
+
+        refresh(indices);
+
+        final String commaSeparatedIndices = Stream.of(indices).collect(Collectors.joining(","));
+
+        final String quotedCommaSeparatedIndices = Stream.of(indices).map((i) -> "\"" + i + "\"").collect(Collectors.joining(","));
+        final String body =
+            "{\"query\":{\"bool\":{\"filter\":[" +
+//                "{\"indices\":{\"indices\":[" + quotedCommaSeparatedIndices + "],\"query\":{\"" + TestDeprecatedQueryBuilder.NAME +  "\":{}}}}," +
```

Comment:
No, I was using that to mess with my deprecation warnings. Removed from the committed code.